alf/algorithms/data_transformer.py

```diff
@@ -363,36 +363,54 @@ def _stack_frame(obs, i):
         for i, field in enumerate(self._fields):
             observation = alf.nest.transform_nest(observation, field,
                                                   partial(_stack_frame, i=i))
-        return experience._replace(observation=observation)
+        return experience.update_time_step_field('observation', observation)
 
 
 class SimpleDataTransformer(DataTransformer):
     """Base class for simple data transformers.
 
     For simple data transformers, there is no state for ``transform_timestep`` and
-    ``transform_timestep`` and ``transform_experience`` use same function
-    ``_transform`` to do the transformation
+    ``transform_experience``. And ``transform_experience`` use the same function
+    ``_transform`` to do the transformation of the ``time_step`` field of the
+    experience.
     """
 
     def __init__(self, transformed_observation_spec):
         super().__init__(transformed_observation_spec, state_spec=())
 
     def transform_timestep(self, timestep: TimeStep, state):
+        """Transform TimeStep.
+        Note that for TimeStep, the shapes are [B, ...].
+
+        Args:
+            timestep (TimeStep): data to be transformed
+        Returns:
+            transformed TimeStep
+        """
         return self._transform(timestep), ()
 
     def transform_experience(self, experience: Experience):
-        return self._transform(experience)
-
-    def _transform(self, timestep_or_exp):
-        """Transform TimeStep or Experience.
+        """Transform Experience.
 
-        Note that for TimeStep, the shapes are [B, ...].
         For Experience, the shapes are [B, T, ...]
 
         Args:
-            timestep_or_exp (TimeStep|Experience): data to be transformed
+            experience (Experience): data to be transformed
```

Comment:
no need to comment type. It's in the function signature.

Reply:
Removed